presto-password-authenticators/src/main/java/io/prestosql/plugin/password/salesforce/SalesforceBasicAuthenticator.java

```diff
@@ -0,0 +1,185 @@
+                "xmlns:urn=\"urn:enterprise.soap.sforce.com\"\n" +
+                "   xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
+                "   xmlns:env=\"http://schemas.xmlsoap.org/soap/envelope/\">\n" +
+                " <env:Header>\n" +
+                "     <urn:CallOptions>\n" +
+                "       <urn:client>presto</urn:client>\n" +
+                "     </urn:CallOptions>\n" +
+                " </env:Header>\n" +
+                " <env:Body>\n" +
+                "   <n1:login xmlns:n1=\"urn:partner.soap.sforce.com\">\n" +
+                "     <n1:username>%s</n1:username>\n" +
+                "     <n1:password>%s</n1:password>\n" +
+                "   </n1:login>\n" +
+                " </env:Body>\n" +
+                "</env:Envelope>\n";
+        String apiVersion = "46.0";
+        String loginUrl = "https://login.salesforce.com/services/Soap/u/";
+        Request request = new Request.Builder()
+                .setUri(URI.create(loginUrl + apiVersion))
+                .setHeader("Content-Type", "text/xml;charset=UTF-8")
+                .setHeader("SOAPAction", "login")
+                .setMethod("POST")
+                .setBodyGenerator(createStaticBodyGenerator(String.format(loginSoapMessage, username, password), UTF_8))
+                .build();
+
+        StringResponseHandler.StringResponse response = httpClient.execute(request, StringResponseHandler.createStringResponseHandler());
+
+        if (response.getStatusCode() != 200) {
+            throw new AccessDeniedException(String.format("Invalid response for login\n.%s",
+                    response.getBody()));
+        }
+
+        Document xmlResponse;
+        try {
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            xmlResponse = builder.parse(new InputSource(new StringReader(
+                    response.getBody())));
+        }
+        catch (ParserConfigurationException | SAXException | IOException e) {
+            throw new RuntimeException(String.format("Error parsing response: %s\n\tReceived error message: %s",
+                    response.getBody(),
+                    e.getMessage()));
+        }
+
+        // Make sure a Session Id has been returned.
+        getElementValue(xmlResponse, "sessionId");
+
+        // We want the organizationId from the response to compare it to the configured org from password-authenticator.properties.
+        String returnedOrg = getElementValue(xmlResponse, "organizationId");
+        // If the only entry in the set is "all", don't bother to check, otherwise make sure the returned org is in the set.
+        // The organizationId is always in Locale.US, regardless of the user's locale and language.
+        if (!(allowedOrganizations.size() == 1 && allowedOrganizations.contains("all"))
```

Comment:
Updating check as follows.

```
// If the only entry in the set is "all", don't bother to check, otherwise make sure the returned org is in the set.
        // The organizationId is always in Locale.US, regardless of the user's locale and language.
        if (!allowedOrganizations.equals(ImmutableSet.of("all"))) {
            if (!allowedOrganizations.contains(returnedOrg.toLowerCase(Locale.US))) {
                throw new AccessDeniedException(String.format(
                        "Login successful, but for wrong Salesforce org.  Got %s, but expected a different org.",
                        returnedOrg));
            }
        }
```